crates/voicevox_core/src/synthesizer.rs

```diff
@@ -504,41 +474,12 @@ mod inner {
             end: usize,
         ) -> Result<Vec<u8>> {
             // TODO: 44.1kHzなどの対応
-            const MARGIN: usize = 14; // 使われているHifiGANのreceptive fieldから計算される安全マージン
-            use std::cmp::min;
-            // 実態(workaround paddingを含まない)上での区間
-            let clipped_start = min(start, audio.frame_length);
-            let clipped_end = min(end, audio.frame_length);
-            // 指定領域が空の区間だった場合、ONNXRuntimeに渡す前に早期リターン
-            if (clipped_start..clipped_end).is_empty() {
-                return Ok(vec![]);
-            }
-            // マージンがデータからはみ出さないことを保証
-            // cf. https://github.com/VOICEVOX/voicevox_core/pull/854#discussion_r1803691291
-            if MARGIN > audio.padding_frame_length + clipped_start
-                || MARGIN > audio.padding_frame_length + (audio.frame_length - clipped_end)
-            {
-                unreachable!("Validation error: Too short padding for input, please report this issue on GitHub.");
-            }
-            let left_margin = MARGIN;
-            let right_margin = MARGIN;
-            // 安全マージンを追加したデータ上での区間
-            let slice_start = audio.padding_frame_length + clipped_start - left_margin;
-            let slice_end = audio.padding_frame_length + clipped_end + right_margin;
-            let segment = audio
-                .internal_state
-                .slice(ndarray::s![slice_start..slice_end, ..]);
+            let spec_segment = crop_with_margin(audio, start..end)?;
             let wave_with_margin = self
-                .render_audio_segment(segment.into_owned(), audio.style_id)
+                .render_audio_segment(spec_segment.to_owned(), audio.style_id)
                 .await?;
-            // 変換前に追加した安全マージンを生成音声から取り除く
-            let wave = wave_with_margin
-                .slice(ndarray::s![
-                    left_margin * 256..wave_with_margin.len() - right_margin * 256
-                ])
-                .into_owned()
-                .into_raw_vec();
-            return Ok(to_s16le_pcm(&wave, &audio.audio_query));
+            let wave = trim_margin_from_wave(&wave_with_margin)?;
+            return Ok(to_s16le_pcm(&wave.to_vec(), &audio.audio_query));
```

Comment:
多分こうでいいはず
(スマホから打ってるのでRustfmtが駄目かも)

```suggestion
            return Ok(to_s16le_pcm(
                wav.to_slice().except("`trim_margin_from_wave` should just trim an array"),
                &audio.audio_query,
            ));
```